cli/medperf/logging/utils.py

```diff
@@ -0,0 +1,274 @@
+    return {
+        "Total Disk Space": disk_usage.total,
+        "Used Disk Space": disk_usage.used,
+        "Free Disk Space": disk_usage.free,
+        "Disk Usage Percentage": disk_usage.percent,
+    }
+
+
+def get_user_information():
+    # Get user information
+    username = getpass.getuser()
+    is_sudoers = "sudo" in [g.gr_name for g in grp.getgrall() if username in g.gr_mem]
+    is_docker_group = "docker" in [
+        g.gr_name for g in grp.getgrall() if username in g.gr_mem
+    ]
+    return {
+        "Username": username,
+        "Is in Sudoers Group": is_sudoers,
+        "Is in Docker Group": is_docker_group,
+    }
+
+
+def get_docker_information():
+    try:
+        exec_path = shutil.which("docker")
+        client = docker.from_env()
+        version = client.version()
+        info = client.info()
+        return {
+            "Docker Installed": True,
+            "Executable path": exec_path,
+            "information": info,
+            "Version": version,
+        }
+    except Exception as e:
+        return {"Docker Installed": False, "Error": str(e)}
+
+
+def get_singularity_information():
+    try:
+        exec_path = shutil.which("singularity")
+        if exec_path is None:
+            return {"Singularity installed": False}
+        conf_path = "/usr/local/etc/singularity/singularity.conf"
+        with open(conf_path, "r") as f:
+            conf = f.readlines()
+        conf_content = []
+        for line in conf:
+            if line.startswith("#") or len(line.strip()):
+                continue
+            conf_content.append(line.strip)
+        config_dict = {}
+        for line in conf_content:
+            key, value = line.split("=")
+            key = key.strip().lower()
+            value = value.strip()
+            if value.isdigit():
+                value = int(value)
+            config_dict[key] = value
+        return {
+            "Singularity installed": True,
+            "Executable path": exec_path,
+            "Configuration": config_dict,
+        }
+    except Exception as e:
+        return {"Singularity installed": False, "Error": str(e)}
+
+
+def get_configuration_variables():
+    config_vars = vars(config)
+    config_dict = {}
+    for item in dir(config):
+        if item.startswith("__"):
+            continue
+        config_dict[item] = config_vars[item]
+    config_dict = filter_var_dict_for_yaml(config_dict)
+    return config_dict
+
+
+def filter_var_dict_for_yaml(unfiltered_dict):
+    valid_types = (str, dict, list, int, float)
+    filtered_dict = {}
+    for key, value in unfiltered_dict.items():
+        if not isinstance(value, valid_types) and value is not None:
+            try:
+                value = str(value)
+            except Exception:
+                value = "<OBJECT>"
+        if isinstance(value, dict):
+            value = filter_var_dict_for_yaml(value)
+        filtered_dict[key] = value
+
+    return filtered_dict
+
```

Comment:
let's just dump the `config.yaml` file in the logs instead of reading `config`?

Reply:
I thought reading the config object would also provide updated values for both the profile and any passed configuraiton argument for a given command. Do you think there would be an issue in using the config object instead of reading the config.yaml file?